packages/extension/src/components/molecules/NetworkIndicator/NetworkIndicator.tsx

```diff
@@ -118,39 +202,132 @@ export const NetworkIndicator: FC = () => {
       {isLoading ? (
         <LoadingOverlay />
       ) : (
-        <Dialog
-          fullScreen
-          open={explanationOpen}
-          onClose={handleClose}
-          TransitionComponent={Transition}
-          data-testid="network-indicator-dialog"
-        >
-          <AppBar sx={{ position: 'relative' }}>
-            <Toolbar>
-              <IconButton edge="start" color="inherit" onClick={handleClose} aria-label="close">
-                <CloseIcon />
-              </IconButton>
-              <Typography sx={{ ml: 2, flex: 1 }} variant="h6" component="div">
-                Change Network
-              </Typography>
-            </Toolbar>
-          </AppBar>
-          <div style={{ overflowY: 'scroll', height: '544px', margin: '20px 20px 0 20px' }}>
-            {Object.keys(NETWORK).map((_network) => {
-              const { name, server, description } = NETWORK[_network as Network];
-              return (
-                <NetworkDisplay
-                  key={_network}
-                  name={name}
-                  server={server}
-                  description={description}
-                  isSelected={name === network}
-                  onClick={() => handleClickOnNetwork(_network as Network)}
-                />
-              );
-            })}
-          </div>
-        </Dialog>
+        <>
+          <AddCustomNetwork
+            dialogOpen={handleAddNetwork}
+            handleClose={handleAddNetworkClose}
+            refreshNetworks={refreshCustomNetworks}
+            networkNames={[...Object.keys(NETWORK), ...Object.keys(existingNetworks)].map((name) =>
+              name.toLowerCase()
+            )}
+          />
+          <Dialog open={confirmDeleteOpen} onClose={() => setConfirmDeleteOpen(false)}>
+            <DialogTitle>Are you sure?</DialogTitle>
+            <DialogContent>
+              <DialogContentText>
+                Are you sure you want to delete the network {networkToDelete}?
+              </DialogContentText>
+            </DialogContent>
+            <DialogActions>
+              <Button onClick={() => setConfirmDeleteOpen(false)} color="primary">
+                Cancel
+              </Button>
+              <Button onClick={handleConfirmDelete} color="primary" autoFocus>
+                Delete
+              </Button>
+            </DialogActions>
+          </Dialog>
+          <Dialog
```

Comment:
This one as well?
This file is quite big

Reply:
Done